project/hooks/pre-commit

```diff
@@ -9,19 +9,19 @@ fi
 
 cd $REPO_ROOT
 
-GOFMT_ERRORS=$(gofmt -s -l . 2>&1)
+GOFMT_ERRORS=$(gofmt -s -l $(find . -type f -name '*.go' -not -path "*/vendor/*") 2>&1)
 if [ -n "$GOFMT_ERRORS" ]; then
 	printf 'gofmt failed for the following files:\n%s\n\nPlease run "gofmt -s -l ." in the root of your repository before committing\n' "$GOFMT_ERRORS" > /dev/stderr
 	exit 1
 fi
 
-GOLINT_ERRORS=$(golint ./... 2>&1)
+GOLINT_ERRORS=$(find . -name '*.go'  -not -path "*/vendor/*" -exec golint {} \;)
```

Comment:
I think this might be slow because it will exec golint for each source file.

What I've usually seen is `golint ./...` and grepping the output to remove lines related to the vendor directory.